bundles/AdminBundle/Controller/Admin/NotificationController.php

```diff
@@ -0,0 +1,222 @@
+ * @license    http://www.pimcore.org/license     GPLv3 and PEL
+ */
+
+declare(strict_types=1);
+
+namespace Pimcore\Bundle\AdminBundle\Controller\Admin;
+
+use Pimcore\Bundle\AdminBundle\Controller\AdminController;
+use Pimcore\Bundle\AdminBundle\Helper\QueryParams;
+use Pimcore\Bundle\AdminBundle\HttpFoundation\JsonResponse;
+use Pimcore\Model\Element\Service;
+use Pimcore\Model\Notification\Service\NotificationService;
+use Pimcore\Model\Notification\Service\NotificationServiceFilterParser;
+use Pimcore\Model\Notification\Service\UserService;
+use Pimcore\Model\User;
+use Pimcore\Translation\Translator;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route("/notification")
+ */
+class NotificationController extends AdminController
+{
+    /**
+     * @param UserService $service
+     * @return JsonResponse
+     * @Route("/recipients")
+     * @Method({"GET"})
+     */
+    public function recipientsAction(UserService $service, Translator $translator): JsonResponse
+    {
+        $this->checkPermission('notifications');
+
+        $data = [];
+
+        foreach ($service->findAll($this->getAdminUser()) as $recipient) {
+            $group = $translator->trans('Group');
+            $prefix = $recipient->getType() == 'role' ? $group . ' - ' : '';
+
+            $data[] = [
+                'id' => $recipient->getId(),
+                'text' => $prefix . $recipient->getName()
+            ];
+        }
+
+        return $this->adminJson($data);
+    }
+
+    /**
+     * @param Request $request
+     * @param NotificationService $service
+     * @return JsonResponse
+     * @Route("/send")
+     * @Method({"POST"})
+     */
+    public function sendAction(Request $request, NotificationService $service) : JsonResponse
+    {
+        $this->checkPermission('notifications');
+
+        $recipientId = (int) $request->get('recipientId', 0);
+        $fromUser = (int) $this->getAdminUser()->getId();
+        $title = $request->get('title', '');
+        $message = $request->get('message', '');
+        $elementId = $request->get('elementId');
+        $elementType = $request->get('elementType', '');
+
+        $element = Service::getElementById($elementType, $elementId);
+
+        if (User::getById($recipientId)) {
+            $service->sendToUser($recipientId, $fromUser, $title, $message, $element);
+        } else {
+            $service->sendToGroup($recipientId, $fromUser, $title, $message, $element);
+        }
+
+        return $this->adminJson(['success' => true]);
+    }
+
+    /**
+     * @param Request $request
+     * @param NotificationService $service
+     * @return JsonResponse
+     * @Route("/find")
+     */
+    public function findAction(Request $request, NotificationService $service): JsonResponse
+    {
+        $this->checkPermission('notifications');
+
+        $id = (int) $request->get('id', 0);
```

Comment:
make sure that one sees only notifications that are assigned to him and cannot set notifications from another user as read.